common/mariadb-galera/helm/templates/storageclass.yaml

```diff
@@ -1,28 +1,58 @@
----
-apiVersion: storage.k8s.io/v1
-kind: StorageClass
-metadata:
-  name: {{ include "commonPrefix" $ }}-cinder
-  labels:
-    {{- include "sharedservices.labels" $ | indent 4 }}
-parameters:
-  type: vmware
-provisioner: cinder.csi.openstack.org
-reclaimPolicy: Delete
-volumeBindingMode: WaitForFirstConsumer
-allowVolumeExpansion: true
----
-apiVersion: storage.k8s.io/v1
-kind: StorageClass
-metadata:
-  name: {{ include "commonPrefix" $ }}-nfs
-  labels:
-    {{- include "sharedservices.labels" $ | indent 4 }}
-parameters:
-  pathPattern: "${.PVC.namespace}-${.PVC.name}"
-  onDelete: "delete"
-  archiveOnDelete: "true"
-provisioner: cluster.local/nfs-subdir-external-provisioner
-reclaimPolicy: Delete
-volumeBindingMode: WaitForFirstConsumer
-allowVolumeExpansion: true
+{{- $cinderEnabled := false }}
+{{- $nfsEnabled := false }}
+{{- range $volumeClaimTemplatesKey, $volumeClaimTemplatesValue := $.Values.volumeClaimTemplates }}
```

Comment:
why not looping over $.Values.volumeMounts instead repeating the conditions 3 times?

Reply:
The [depth is not the same](https://github.com/sapcc/helm-charts/blob/0e874cc3d35dbe4d19901e8c7aea960929d7b113/common/mariadb-galera/helm/values.yaml#L1044).
```yaml
volumeMounts:
  database:
    data:
    log:
  proxy:
    proxysql:
      data:
  backup:
    kopia:
      data:
```
We should refactor [volumeMounts](https://github.com/sapcc/helm-charts/blob/0e874cc3d35dbe4d19901e8c7aea960929d7b113/common/mariadb-galera/helm/values.yaml#L1044) and [volumeClaimTemplates](https://github.com/sapcc/helm-charts/blob/0e874cc3d35dbe4d19901e8c7aea960929d7b113/common/mariadb-galera/helm/values.yaml#L1095C1-L1095C21) by merging them into one option set. This means less flexibility because you than cannot reuse a volumeClaimTemplate anymore, but this is anyhow not used currently.